Frontier silicon

zhelev opened the pull request:
**Description:**
new component - frontier silicon, 'depends on latest' netdisco

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#1966

**Example entry for `configuration.yaml` (if applicable):**
```yaml
media_player:
  - platform: frontier_silicon
    host: IP_ADDRESS
```

balloob committed:
Merge pull request #5530 from home-assistant/dev

0.37

DavidLP committed:
Fixes: Pilight Switch rejects alphanumeric IDs #5119 (#5601)

Danielhiversen committed:
bug fix in hue (#5623)

tchellomello committed:
Fixes issue #5627 by bumping external Amcrest module to version 1.1.4 (#5662)

Add an optional extended description…

molobrakos committed:
Upgraded tellduslive (#5664)

pvizeli committed:
Bugfix sonos group coordinator (#5691)

* Bugfix sonos group coordinator

* Fix tests

pvizeli committed:
Bugfix async blocking loop with xml parser. (#5694)

jabesq committed:
Hotfix for netatmo cameras (#5644)

* Fix for missing netatmo tags in 0.37

Also fix issue with SSL certificate for vpn_url

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Netatmo welcome: vpn_url can be empty

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* add config floag to disable SSL verification for vpn_url

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Import CONF_VERIFY_SSL from const

pvizeli committed:
Fix bug for UNREACH devices / Variable handling and update. (#5689)

* Fix bug for UNREACH devices / Variable handling and update.

* fix track_time

* update after data after creation

* add message output

* change unreach

* change unreach code

* Revert "change unreach code"

This reverts commit f58430de3cb854d5f6f886bfe4d59899c1efadbf.

* update pyhomematic

pvizeli committed:
Bugfix sonso source input (#5699)

turbokongen committed:
[lock.zwave] Bugfix Zwave lock  (#5619)

* Bugfix state

* remove debug

aequitas committed:
Prevent infinite loop in crossconfigured mqtt event streams (#5624)

* Prevent events about MQTT messages received to cause infinite loop when two HA instances are crossconfigured for mqtt_eventstream.

* Fix linting

* Publish all MQTT received events except incoming from eventstream. Also make it configurable.

tboyce021 committed:
Fix hue lightgroups not syncing state (#5702)

balloob committed:
Version bump to 0.37.1

balloob committed:
Merge pull request #5707 from home-assistant/release-0-37-1

0.37.1

zhelev committed:
added frontier_silicon constant

zhelev committed:
added frontier_silicon constant

zhelev committed:
added frontier_silicon constant

zhelev committed:
added the frontier_silicon component

zhelev committed:
added the frontier_silicon component

zhelev committed:
added the frontier_silicon component

zhelev committed:
added the frontier_silicon component

zhelev committed:
added the frontier_silicon component

zhelev committed:
added the frontier_silicon component

zhelev committed:
cleaning up according to travis

zhelev committed:
cleaning up according to travis

zhelev committed:
cleaning up according to travis

zhelev committed:
cleaning up according to travis

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
trying to satisfy pylint

zhelev committed:
fsapi version 0.0.6

zhelev committed:
fsapi version 0.0.6

zhelev committed:
fsapi version 0.0.6

zhelev committed:
fsapi version 0.0.6

zhelev committed:
fsapi version 0.0.6

zhelev committed:
fsapi version 0.0.6

zhelev committed:
with fsapi version 0.0.7

zhelev committed:
with fsapi version 0.0.7

zhelev committed:
with fsapi version 0.0.7

zhelev committed:
with fsapi version 0.0.7

zhelev commented:
I need some help here (my first python project) - the travis requirements check won't pass. 
I have a single requirement REQUIREMENTS = ['fsapi==0.0.7'] (https://pypi.python.org/pypi/fsapi/0.0.7). 
I guess I did sth. wrong in the _fsapi_ project: https://github.com/zhelev/python-fsapi. 
The project requires _lxml_ and _requests_ itself. 
The components works as a custom component. Any ideas what I did wrong?

scarface-4711 commented:
hey @zhelev 
you need to add something like 
`# homeassistant.components.media_player.frontier_silicon`
`fsapi==0.0.7`
at the right place of **requirements_all.txt** in home-assistant root directory

scarface-4711 commented on homeassistant/components/media_player/frontier_silicon.py:
```diff
@@ -0,0 +1,261 @@
+    def get_fs(self):
+        """ Create a fsapi session."""
+        from fsapi import FSAPI
+        return FSAPI(self._device_url, self._password)
```
afaik this will block the event loop as FSAPI includes a HTTP request
I guess you should either define get_fs() as a coroutine and yield the FSAPI call or leave get_fs() as it is, but yield it each time when it is called.

zhelev committed:
added fsapi dependency

zhelev committed:
added fsapi dependency

zhelev committed:
added fsapi dependency

zhelev committed:
added fsapi dependency

zhelev committed:
yielding the FSAPI

zhelev committed:
yielding the FSAPI

zhelev committed:
yielding the FSAPI

zhelev committed:
yielding the FSAPI

zhelev committed:
Removing white space from docstring

zhelev committed:
Removing white space from docstring

zhelev committed:
Removing white space from docstring

zhelev committed:
Removing white space from docstring

zhelev committed:
Removing white space from an empty line

zhelev committed:
Removing white space from an empty line

zhelev committed:
Removing white space from an empty line

zhelev committed:
Removing white space from an empty line

scarface-4711 commented:
@zhelev: 
I just had a look at your library and I guess it would be better to include is as a non asyncio platform, as it does not look async friendly.
Maybe someone with good experience in asynchronous programming like @balloob can confirm or rebut my impression.

zhelev committed:
Switching to sync

zhelev committed:
Switching to sync

zhelev committed:
Switching to sync

zhelev committed:
Switching to sync

zhelev committed:
clean up white spaces and rename device to FSAPIDevice

zhelev committed:
clean up white spaces and rename device to FSAPIDevice

zhelev committed:
clean up white spaces and rename device to FSAPIDevice

zhelev committed:
clean up white spaces and rename device to FSAPIDevice

zhelev commented:
@scarface-4711 Thanks for the feedback, appreciate it. I've reverted to a sync implementation for now and am still trying to get the requirements to work.

zhelev committed:
changed info to debug

zhelev committed:
changed info to debug

zhelev committed:
Merge branch 'frontier_silicon' of https://github.com/zhelev/home-assistant into frontier_silicon

zhelev committed:
merging

zhelev committed:
remove white space

JshWright commented:
@zhelev In the root of your repo, run `script/gen_requirements_all.py`. That will update the requirements file (then commit the resulting changes).

zhelev committed:
generated requirements

zhelev committed:
generated requirements

zhelev committed:
Merge branch 'dev' into frontier_silicon

pvizeli committed:
Ffmpeg update 1.4 (#5813)

* Pump ffmpeg version

* update entity

* next 1.4

balloob commented on homeassistant/components/media_player/frontier_silicon.py:
```diff
@@ -0,0 +1,247 @@
+    password = config.get(CONF_PASSWORD)
+
+    try:
+        if host and port and password:
```
No need for this if you use `vol.Required` instead of `vol.Optional` in PLATFORM_SCHEMA.

balloob commented:
It's too bad that you removed the checklist from your API because besides the requirements instruction that you missed, you also missed the coverage task. Please do that one too.

Teagan42 committed:
Change medium state for filtering (#5817)

sdague committed:
Enhancements to ARWN platform (#5816)

* Fix arwn platform discover_sensors

The discover_sensors function can return either singletons or a list
of sensors. However the consumer was always expecting a list. This
fixes it to work in both cases.

* Add custom icons to arwn sensors.

This adds some custom icons for different kinds of weather sensors
that the arwn platform returns. Makes it a little easier to see what's
going on.

Commit:
Fixed call to object_id() (#5814)

kellerza committed:
[recorder] Protect against running in the event loop (#5812)

turbokongen committed:
Bugfixes (#5740)

* wrong data for lock alarm_type

* missing whitespace

* Not possible to set codes starting with 0

Duoxilian committed:
Support away_mode as permanent hold and hold_mode as temporary hold. (#5725)

* Support away_mode as permanent hold and hold_mode as temporary hold.

* Add comments to explain code better. Remove indefinite hold preference
to be consistent with 'away_mode'.

balloob committed:
Remove non-working webfont preload (#5819)

balloob committed:
Update frontend (#5820)

balloob committed:
Upgrade aiohttp to 1.3 (#5821)

pvizeli committed:
Move signal handling out of core to bootstrap (#5815)

* Move signal handling out of core to bootstrap

* Fix tests

balloob committed:
Deprecate event decorators (#5822)

balloob committed:
Make device sun light trigger async (#5823)

turbokongen committed:
Adding helper for get and set values (#5743)

* cleanup

* Update __init__.py

* Update __init__.py

* Update __init__.py

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

molobrakos committed:
Don't thow exception if connection to server is lost (#5775)

balloob committed:
Convert config.components to a set (#5824)

postlund committed:
Handle connection errors when connecting to Apple TVs (#5829)

* Handle connection errors when connecting to Apple TVs

Also bump pyatv to 0.1.2 which fixes a request leak.

* Fix pylint error

* Fix import order

scipioni committed:
Telegram webhooks (#5793)

* new component telegram_webhooks

* keyboard support in telegram notify

* telegram_webhooks has no tests

* requirement like notify/telegram

* ops, requirements_all.txt needed for travis

* ops, requirements_all.txt is generated by script/gen_requirements_all.py

* check telegram trusted networks in web handler

* raise an event now

* use of hass.config.api.base_url

* more readable

* small cleanups

* Small style change for HA guideline

* fix lint

* revert return to origin

postlund committed:
Reuse default aiohttp session (#5836)

pvizeli committed:
Update aiohttp 1.3.1 (#5838)

pschmitt committed:
Refactoring and JSON decode error handling (#5826)

* Refactoring and JSON decode error handling

* Catch ValueError instead of simplejson.scanner.JSONDecodeError

emlove committed:
[recorder] Add tests for full schema migration (#5831)

* [recorder] Add tests for full schema migration

* Remove leftover code

* Fix duplicate creation of sqlalchemy Index object

* It's that kind of day...

* Improve models_original docstring

bokub committed:
Add history_stats sensor (#5610)

* Add new sensor/history_stats component

* Add real unit tests

* Code style changes requested for pull request

* Remove time aliases & allow datetime values

* Reformat unit tests

* Remove all async behavior in history_stats

* Change duration format

markoudev committed:
Add support for multiple devices to Tado device tracker (#5810)

* Add support for multiple devices to Tado device tracker

* Hound did not agree with my usage of spaces

* Two blank lines were expected

* Keep CONF_HOME_ID in the Tado file

* Make home_id optional

* Work with different API URLs depending on home_id being present

* Remove CONF_HOME_ID from HA's const.py file

* Missed removing CONF_HOME_ID from the import

andrey-git committed:
Fix zwave helper getter not to fail on some None results. (#5845)

LinuxChristian committed:
D-Link switch version bump of external library (#5843)

rytilahti committed:
bump python-yeelight version (#5850)

Add an optional extended description…

pvizeli committed:
telegram small cleanup/style (#5852)

pvizeli committed:
Fix check_config script. (#5853)

andrey-git committed:
Add support for zwave wakeup setting. (#5849)

* Add support for zwave wakeup setting.

* rename wakeup

molobrakos committed:
Do not call state if device isn't available (#5835)

balloob committed:
Core: cleanup timer (#5825)

* Minor core cleanup

* Cleanup timer

* Lint

* timeout with correct loop

* Improve timer thanks to pvizeli

* Update core.py

* More tests

balloob committed:
Update frontend (#5855)

kellerza committed:
[recorder] Run end model changed in session scope (#5858)

balloob committed:
Recorder run can be None (#5854)

zhelev committed:
rere-merged files

zhelev committed:
pylint

zhelev commented:
@balloob I am pretty new here and it is my first python project. Can you give me a hint what do you mean with removing the 'checklist'?  So far I just followed the tips from the other guys and updated my branch. 
The coverage fails because I do not have a unit test or there is any another reason?

zhelev committed:
moved import requests to the method where it is being used

zhelev committed:
add a basic unit test

zhelev committed:
cleaned up source code

zhelev commented:
I guess I've reached a dead end here, closing the PR and will retry, will revisit all steps and hope this time it works out
